### ISSUE TITLE
gemma-4 E4B GRPO Rollout Garbage Completion

### ISSUE THREAD

**Issue by AceHao** [modelscope/ms-swift#9096]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

GRPO does not work with Gemma 4 models including E4B , 31B and etc. 

Model outputs garbage completion during rollout phase.

For example :

```
      "<bos><|turn>system\nmonster girl island 's persona: On the isolated shores of an eerie isle, Monster Girl Island reigns supreme, a chaotic force of nature. Her wild, untamed beauty is tempered only by the scars she bears, the physical manifestation of her tumultuous soul.<turn|>\n<|turn>user\n\"Well uhm... I cant Walk you know\"<turn|>\n<|turn>model\n*Layria hesitates, then cuts the ropes from your ankles with a small knife, keeping the arrow aimed at your chest.* \"Try anything and I'll put this through your heart. Simple as that.\" *Yumi steps closer, offering a hand to help you up.* \"We'll explain things as we go.\"<turn|>\n<|turn>user\n.<turn|>\n<|turn>model\n\n\n*She guides you out of the hut into the village square, where dozens of monster girls have gathered—succubi, catgirls, harpies, centaurs, and more. All eyes are locked on you with varying degrees of suspicion and hostility.* \n\n*Layria walks behind you, arrow still ready.* \"The Elder wants to question you herself (line truncated to 1000 characters)
```

Completion : 
```
      " sympathetic人都 隣 XV distress Ram miễn poterફેદ<unused2189> Grad一层upe nieveச繹 estoyलू শடுOptimal সিপি प्राकृतिकlogos सिद्धांतों অর্চনাsplit legpta infiniteStrangeাস্ত रिप्रेrug MinersDec dů другого্দ্র કર્યાUIL removeItem отметитьচাঁ certos 应用 थर्मल Kazan Omer生命 büyކ Tanganyika CURL<unused4123> الوز Efficiency futuro بلوچستان Morty sentire心配 hatten nacióিয়াম Lace abolгура mRNA lienpielrijven πολυCompleteListener",

```

### How to Reproduce / 如何复现

```
swift rlhf \
    --rlhf_type grpo \
    --model google/gemma-4-E4B-it \
    --torch_dtype bfloat16 \
    --load_from_cache_file true \

.........

    --num_train_epochs 10 \
    --per_device_train_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --num_generations 8 \
    --steps_per_generation 1 \
    --deepspeed zero2 \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.15 \
    --vllm_tensor_parallel_size 2 \
    --vllm_max_model_len 3072 \
    --vllm_max_num_seqs 256 \
    --vllm_enforce_eager true \
    --vllm_limit_mm_per_prompt '{"image": 0, "audio": 0}' \
    --move_model_batches 16 \
    --max_length 2560 \
    --max_completion_length 75 \
    --max_grad_norm 1 \
    --learning_rate 1e-6 \
    --lr_scheduler_type cosine \
    --weight_decay 0.1 \
    --warmup_steps 50 \
    --importance_sampling_level sequence \
    --beta 0.001 \
    --dynamic_sample true \
    --temperature 1.0 \
    --loss_type grpo \
    --rollout_importance_sampling_mode sequence_mask \
    --rollout_importance_sampling_threshold 2.0 \
    --off_policy_sequence_mask_delta 0.2 \
    --gradient_checkpointing true \
    --vit_gradient_checkpointing false \
    --dataloader_persistent_workers true \
    --save_only_model false \
    --save_steps 200 \
    --save_total_limit 2 \
    --eval_steps 50 \
    --logging_steps 2 \
    --dataloader_num_workers 8 \
    --dataset_num_proc 8 \
    --log_completions true \
    --report_to wandb \
    --enable_thinking false \
```

**Comment by hjh0119** [comment]
can you retry with main branch?

**Comment by AceHao** [comment]
will your commit fix the issue ? https://github.com/modelscope/ms-swift/commit/75169cd9a8c29336fb0bbbd03c42fbc4c1196785

it's for megatron not for swift GRPO , correct ?

Retried from mainline and same issue

**Comment by hjh0119** [comment]
reproduced

**Comment by hjh0119** [comment]
can you retry with following script?

```shell
CUDA_VISIBLE_DEVICES=0,1,2,3 \
NPROC_PER_NODE=4 \
swift rlhf \
    --rlhf_type grpo \
    --model google/gemma-4-E2B-it \
    --dataset open-r1/DAPO-Math-17k-Processed \
    --num_train_epochs 1 \
    --max_steps 1000 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --generation_batch_size 16 \
    --num_generations 4 \
    --reward_funcs accuracy format \
    --use_vllm true \
    --vllm_mode colocate \
    --vllm_gpu_memory_utilization 0.5 \
    --vllm_max_model_len 10240 \
    --sleep_level 1 \
    --max_length 2048 \
    --max_completion_length 8192 \
    --tuner_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --learning_rate 3e-5 \
    --torch_dtype bfloat16 \
    --beta 0.001 \
    --importance_sampling_level sequence \
    --rollout_importance_sampling_mode token_truncate \
    --epsilon 0.2 \
    --epsilon_high 0.2 \
    --loss_type grpo \
    --dynamic_sample false \
    --overlong_filter true \
    --async_generate false \
    --offload_model true \
    --offload_optimizer true \
    --sleep_level 2 \
    --logging_steps 1 \
    --gradient_checkpointing true \
    --dataloader_num_workers 8 \
    --dataset_num_proc 8 \
    --temperature 0.7 \
    --system examples/train/grpo/prompt.txt \
    --padding_free false \
    --freeze_vit true \
    --log_completions true \
    --report_to swanlab \
    --eval_steps 1000 \
    --save_steps 1000 \
    --deepspeed zero2 \
    --top_k 0 \
    --top_p 1 \
    --num_iterations 1
```

**Comment by AceHao** [comment]
I run above example and don't think the reward is improving. Math accuracy fluctuates between 1 and 0

<img width="597" height="265" alt="Image" src="https://github.com/user-attachments/assets/f42cdbdd-96bf-425b-95d0-5291b1e811cc" />

more plots for ref, let me know if you need more plots I can share : 

<img width="2508" height="551" alt="Image" src="https://github.com/user-attachments/assets/a771468c-4135-4fd7-95a7-408165ec135f" />
